hypershift-operator/controllers/nodepool/nodepool_controller.go

```diff
@@ -934,13 +934,17 @@ func validateConfigManifest(manifest []byte) error {
 	return nil
 }
 
-func (r *NodePoolReconciler) getReleaseImage(ctx context.Context, releaseImage string) (*releaseinfo.ReleaseImage, error) {
+func (r *NodePoolReconciler) getReleaseImage(ctx context.Context, hostedCluster *hyperv1.HostedCluster, releaseImage string) (*releaseinfo.ReleaseImage, error) {
+	pullSecret := &corev1.Secret{}
+	if err := r.Client.Get(ctx, client.ObjectKey{Namespace: hostedCluster.Namespace, Name: hostedCluster.Spec.PullSecret.Name}, pullSecret); err != nil {
+		return nil, fmt.Errorf("cannot get pull secret %s/%s: %w", hostedCluster.Namespace, hostedCluster.Spec.PullSecret.Name, err)
+	}
 	ReleaseImage, err := func(ctx context.Context) (*releaseinfo.ReleaseImage, error) {
 		ctx, span := r.tracer.Start(ctx, "image-lookup")
 		defer span.End()
 		lookupCtx, lookupCancel := context.WithTimeout(ctx, 1*time.Minute)
 		defer lookupCancel()
-		img, err := r.ReleaseProvider.Lookup(lookupCtx, releaseImage)
+		img, err := r.ReleaseProvider.Lookup(lookupCtx, releaseImage, pullSecret.Data[corev1.DockerConfigJsonKey])
```

Comment:
Should probably validate the key exists to fail fast and avoid passing through a `nil` in the case of a malformed secret

Reply:
added check